lib/authenticator.js

```diff
@@ -1,37 +1,38 @@
-'use strict'
+import GitLab from './api';
+import error from './error';
 
-const GitLab = require('./api')
-const error = require('./error')
-
-class Authenticator {
+export default class Authenticator {
   constructor (opts) {
     opts = opts || {}
-    this.api = opts.api || new GitLab(opts)
+    // this.api = opts.api || new GitLab(opts)
+    this.api = opts.api || new GitLab(opts);
   }
 
-  authenticate (credentials, cb) {
-    if (!valid(credentials)) return error.defer(cb, 500, 'Invalid credentials')
-    let self = this
-    let username = credentials.body.name
-    let email = credentials.body.email
-    self.api.login(username, credentials.body.password, email)
-      .then((token) => {
-        cb(null, {
-          token: token,
-          user: {
-            name: username,
-            email: email || username + '@devnullmail.com'
-          }
-        })
-      })
-      .catch((err) => {
-        cb(err)
-      })
+  async authenticate (credentials, cb) {
+    if (!valid(credentials)) return error.defer(cb, 500, 'Invalid credentials');
+    try {
+      const request = await this.api.users();
```

Comment:
Rather than query all users, which requires an access token with `sudo` / `admin` permissions please query the current user. If the request returns 401 or 403 then the user is not authenticated. 

https://docs.gitlab.com/ee/api/users.html#user

Reply:
This should be doable with `api` permissions though may also require `read_user` permissions. This is still better than reading all users.